### ISSUE TITLE
A moderator who is banned on another instance does not have their moderation actions federate to that other instance.

### ISSUE THREAD

**Issue by SomethingElseForAChange123** [LemmyNet/lemmy#4409]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

A moderator who is banned on another instance does not have their moderation actions federate to that instance. This makes properly moderating impossible.

### Steps to Reproduce

1. User A becomes a moderator of a community on instance X
2. User A gets site-wide banned on instance Y
3. User B posts rule breaking stuff on A's community
4. User A removes B's stuff
5. User B's rule breaking posts remain in place on instance Y

### Technical Details

https://lemm.ee/comment/8725321 has bans and removals in place, created by Throwaway who is banned on lemmy.world

https://lemmy.world/comment/7072191 does not have these bans and removals.

### Version

BE: 0.19.3

### Lemmy Instance URL

lemm.ee

**Comment by Nutomic** [comment]
A site ban means that all activities from that user will be rejected so this is expected. Only solution I can imagine is to ignore site bans (and also community bans?) when receiving mod actions. But Im not sure thats a good idea, the user could perform malicious mod actions and it would be very confusing that the ban would seem to be ignored.

**Comment by dessalines** [comment]
Agree, someone could do malicious mod actions if we allowed this.

**Comment by sunaurus** [comment]
> The moderation should only federate if Community A is on the same instance than the moderator ( that got site banned from the instance )

This makes sense to me, any objections to this approach @dessalines or @Nutomic ?

**Comment by Nothing4You** [comment]
> Agree, someone could do malicious mod actions if we allowed this.

This argument doesn't make sense to me.
Maybe I'm also misunderstanding what remote communities are supposed to be.

My understanding is currently that a remote community, as it is displayed on my instance, is supposed to be a cached representation of the actual remote community, possibly with some content removed (moderated) locally or rejected due to banned users.
Derived from this, content that is removed on the community's home instance should always be removed on all other instances, including my instance, unless explicitly restored by an admin of that instance.
This is loosely related to what I described in https://github.com/LemmyNet/lemmy/issues/4118#issuecomment-1784888701, where all other moderators for this community will see the content removed already and won't re-remove it to "fix federation".

I can't think of a good reason where preventing "malicious mod actions", which would also only work if the user is already banned prior to those actions, would be weighed higher than having spam and other bad content removed as a result of a moderation action.

**Comment by SomethingElseForAChange123** [comment]
Yeah, I would love an example of a malicious mod action. Maybe pinning a malicious post, but in that case that post should covered by the ban.

**Comment by Nutomic** [comment]
> The moderation should only federate if Community A is on the same instance than the moderator ( that got site banned from the instance

That seems insufficient, because Lemmy also allows users to moderate communities on other instances. So we would be back to allowing all mod actions from banned users. Which would still be incomplete as announcement posts or comments with warnings would remain blocked. Anyway the relevant code is [in these functions](https://github.com/LemmyNet/lemmy/blob/9367cbdb009c9a2a1658ceeb74ceec20fccaf6bc/crates/apub/src/activities/mod.rs#L77-L125).

**Comment by DraconicNEO** [comment]
> Yeah, I would love an example of a malicious mod action. Maybe pinning a malicious post, but in that case that post should covered by the ban.

Also for malicious mod actions those are things that the home instance or mods of that community have the responsibility to deal with, by removing that user as a mod from their community.
Excluding their "bad faith" mod actions from one instance doesn't make sense when they will otherwise affect the rest of the fediverse.
In that regard it would make sense for moderator actions to still federate and be accepted, because ignoring removals and bans is irresponsible because it allows bad faith actors to run wild there.

So with this in mind it is best to allow mod actions that aren't posts or comments to federate. It should be up to an instance to deal with their bad faith moderators, and the other instance to simply remove that Community if the home instance of that Community does not wish to comply.
For moderators that moderate from a different instance if the community that they are moderating is on the instance they are banned from, they should simply be removed as a mod there, not ignored that is what people would call a shadow ban. If the community is not homed on the instance they are banned from, actions should resume normally with posts and comments being ignored and excluded.

**Comment by DraconicNEO** [comment]
Was this issue fixed? Mod actions now do seem to federate even when the moderator is site banned.

**Comment by Nutomic** [comment]
There has been no related change.

**Comment by DraconicNEO** [comment]
> There has been no related change.

Oh, interesting. I noticed that mods who were banned from an instance still have their mod actions federate to the instance that banned them (post and comment removals and bans appear in the communities of the instance that banned them, and show in the modlog).

**Comment by dessalines** [comment]
If they're only community banned, then I think their mod actions will still federate fine.

**Comment by DraconicNEO** [comment]
> If they're only community banned, then I think their mod actions will still federate fine.

They're banned from the other instance. Mod and community are on lemmy.zip and they're banned from lemmy.blahaj.zone, post & comment removals, as well as bans still seem to be going through to lemmy.blahaj.zone even though they're banned on the instance.

**Comment by DraconicNEO** [comment]
Just found [another, more clear example of this in the wild](https://lemmy.dbzer0.com/comment/24273451) on my instance the lead and only mod of the community is site banned and has been banning or removing content in bad faith, and the actions never federate to our server leading people to not think they are banned, yet the actions federate elsewhere. This creates a sort of unintended Shadowban that still is effective but none of us know about it unless digging deeper. Which isn't great.

I would propose that in a future version if it hasn't already, that mod actions be exempt from the site ban restriction. Especially because in cases where communities like this can become toxic cesspools without the mod's actions federating to that server. And it doesn't make a difference to not accept them on our server when they are accepted by every other. It only serves to (unintentionally) shadowban our users from that community.
